### DISCUSSION TITLE
Correctly reference backend in frontend

### DISCUSSION THREAD

**Discussion by UrielMhezzek** [microsoft/aspire#1742] [Q&A]
I have an existing backend api. This should now get a new frontend. The fontend supports both render modes server-side and client-side. I have created a normal web project for this purpose. This then added two projects to me. the frontend for the server and a web assembly for the client. if I adjust the program.cs in the client and add 

`builder.Services.AddScoped(sp => new HttpClient { BaseAddress = new Uri($"http://backend") });` 

the httpclient does not connect to the backend but "backend" is not replaced by aspiere. How can I change this?

**Comment by petermorlion** [comment]
Not sure if this helps, but as you're in the Aspire issues, I'm assuming you're using Aspire and then you could do this:

```
var builder = DistributedApplication.CreateBuilder(args);

var backend = builder.AddProject<Projects.Backend_Api>("backend-api");

builder.AddProject<Projects.Frontend_Api>("frontend")
    .WithEnvironment("backend-url", backend.GetEndpoint("https"));

builder.Build().Run();
```

This goes in the `Program.cs` of your Aspire AppHost. Then you can retrieve the base url of the backend in your frontend app:

```
var baseUrl = System.Environment.GetEnvironmentVariable("backend-url");
```
